### PR TITLE
apps: add mac, cipher and digest arguments to the kdf applet.

### PR DESCRIPTION
**Pull request by paulidale**
This adds -digest, -mac and -cipher options to reduce the use of -kdfopt

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: apps/kdf.c [hunk 1]
```diff
@@ -83,6 +106,21 @@ int kdf_main(int argc, char **argv)
...
+        case OPT_DIGEST:
+            digest = alloc_kdf_algorithm_name(&opts, "digest", opt_arg());
+            if (digest == NULL)
+                goto opthelp;
+            break;
+        case OPT_MAC:
+            mac = alloc_kdf_algorithm_name(&opts, "mac", opt_arg());
+            if (mac == NULL)
+                goto opthelp;
+            break;
```
Comment (t8m) [inline apps/kdf.c:123]: If any of these options are used multiple times, it will be leaked.

**Comment by levitte** [comment]
Side note: "applet" has my brain switch over to Java-think...  imagine the confusion 😆

File: apps/kdf.c (hunk 1 above)
Comment (paulidale) [reply apps/kdf.c]: Fixed.

**Review by t8m — APPROVED** [review]

**Comment by paulidale** [comment]
Merged to master, thanks for the feedback and comments.

**Pull request closed without merging**